### DISCUSSION TITLE
Sales Order Header Missing and Page Format Messed Up

### DISCUSSION THREAD

**Discussion by flecb001** [OpenMage/magento-lts#2830] [Questions & Answers]
Hi. I'm pretty unfamiliar with OpenMage/Magento but have been handed a problem to solve. On my sales_order/view page the header at the top that contains the order # and date time, along with the buttons for Credit Memo, Refulfill, Resubmit, etc are missing. The interesting thing is that of 8 environments (6 dev and 2 prod) that this code is running in the header is there in all except one. Exact same code can be connected to each of the environments so I'm pretty sure it's not actually the files causing an issue, but who knows. Perhaps something in the database but I'm not seeing anything in there that makes sense. I'm looking for suggestions on how I could find the issue. I'm reading up on the file structure but this is pretty foreign compared to what I normally code in.

![image](https://user-images.githubusercontent.com/42824721/208793506-4d7d8b86-491a-49ec-bf6f-9ce73244088e.png)

With the header missing (not even present in the HTML and hidden) the details get tossed into the left side column.

![image](https://user-images.githubusercontent.com/42824721/208793696-1ef4b546-77a6-40fe-8fbe-055bc6d457ae.png)

Any thoughts or suggestions would be appreciated.

**Comment by kiatng** [comment]
It looks like the page is backend. The appearance of buttons  _Credit Memo_, _Refulfill_, _Resubmit_ depend on the order. And it looks like there is/are extension(s) rendering the buttons.  In the core, the rendering happens in this file:

https://github.com/OpenMage/magento-lts/blob/355cff5a07cfddfb3822df6003b742d13e352052/app/code/core/Mage/Adminhtml/Block/Sales/Order/View.php#L43-L197

Look for `$this->_removeButton()` and `$this->_addButton()`. As you can see, the buttons are added depending on certain conditions.

The extension(s) that add additional buttons should be located in these folders: 
1. app\code\local\
2. app\code\community

To locate the file(s), do a search for `Refilfill` or `$this->_addButton` in the above folders.

**Comment by kiatng** [comment]
The mess up style on the page is probably due to an error/exception. Check for errors in `var/log/exception.log` and `var/log/system.log`. If the developer mode is enabled (at the server config):
https://github.com/OpenMage/magento-lts/blob/355cff5a07cfddfb3822df6003b742d13e352052/app/Mage.php#L28-L33

the error message can sometimes be hidden on the order page. You can look for it by pressing F12 in your browser to examine the HTML source.
